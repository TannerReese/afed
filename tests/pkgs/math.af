# Test num library

# Constants
pi = ``, math.pi = ``,
e = ``, math.e = ``,
gold = ``, math.gold = ``,

# Functions print themselves
ceil = ``, asinh = ``,

# Piecewise functions
abs (1 / 2) + math.abs (-4 / 5) = ``,
abs (-4.38) - math.abs 8.9 = ``,
math.signum 67.243 + 10 * signum (-5 / 3) = ``,
real (450 / 23) = ``,
floor (7843 / 193) - floor (-89.82) = ``,
ceil 29.8493 * ceil (28 % (17/2)) = ``,
round 2.354 + round (31 / 2) = ``,

# Real functions
sqrt (32 / 12) = ``,
math.sqrt (-1) = ``,
cbrt (12 / 83) = ``,

# Trig functions
sin 3.2 = ``, cos 1.1 = ``, tan 4.3 = ``,
sinh 9.7 = ``, cosh 2.2 = ``, tanh 0.7 = ``,
asin 0.1651 = ``, acos 0.212 = ``, atan 32.1 = ``,
asinh (-23.4) = ``, acosh 12.1 = ``, atanh (-0.42) = ``,
atan2 3.45 (-4.23) = ``,

# Exponentials & Logarithms
exp 3 = ``,
exp2 (3/2) = ``,
ln 25 = ``, log10 9 = ``, log2 8.4 = ``,
log 7 10 = ``,

# Digits in any base
17493 .digits 13 = ``,

# Combinatorial functions
gcd (69 / 34) (111 / 106) = ``,
lcm (6 / 91) (52 / 17) = ``,

factorial 12 = ``,
choose 2.34 7 = ``,
choose (21 / 19) 3 = ``,
choose 43 30 = ``,

