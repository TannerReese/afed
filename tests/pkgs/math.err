line 22, column 18 of "math.af" Eval Error: Cannot take square root of negative
1 Eval Error encountered
