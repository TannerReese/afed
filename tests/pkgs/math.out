# Test num library

# Constants
pi = `3.141592653589793`, math.pi = `3.141592653589793`,
e = `2.718281828459045`, math.e = `2.718281828459045`,
gold = `1.618033988749895`, math.gold = `1.618033988749895`,

# Functions print themselves
ceil = `math.ceil`, asinh = `math.asinh`,

# Piecewise functions
abs (1 / 2) + math.abs (-4 / 5) = `13 / 10`,
abs (-4.38) - math.abs 8.9 = `-4.5200000000000005`,
math.signum 67.243 + 10 * signum (-5 / 3) = `-9`,
real (450 / 23) = `19.565217391304348`,
floor (7843 / 193) - floor (-89.82) = `130`,
ceil 29.8493 * ceil (28 % (17/2)) = `90`,
round 2.354 + round (31 / 2) = `18`,

# Real functions
sqrt (32 / 12) = `1.632993161855452`,
math.sqrt (-1) = `Eval Error: Cannot take square root of negative`,
cbrt (12 / 83) = `0.5248490126692728`,

# Trig functions
sin 3.2 = `-0.058374143427580086`, cos 1.1 = `0.4535961214255773`, tan 4.3 = `2.28584787736698`,
sinh 9.7 = `8158.803568365963`, cosh 2.2 = `4.567908328898228`, tanh 0.7 = `0.6043677771171636`,
asin 0.1651 = `0.1658594018606022`, acos 0.212 = `1.357175301648401`, atan 32.1 = `1.5396537507072512`,
asinh (-23.4) = `-3.846339461524289`, acosh 12.1 = `3.184640709347431`, atanh (-0.42) = `-0.44769202352742066`,
atan2 3.45 (-4.23) = `2.4574099318598117`,

# Exponentials & Logarithms
exp 3 = `20.085536923187668`,
exp2 (3/2) = `2.8284271247461903`,
ln 25 = `3.2188758248682006`, log10 9 = `0.9542425094393249`, log2 8.4 = `3.070389327891398`,
log 7 10 = `1.1832946624549385`,

# Digits in any base
17493 .digits 13 = `[8, 6, 12, 7]`,

# Combinatorial functions
gcd (69 / 34) (111 / 106) = `3 / 1802`,
lcm (6 / 91) (52 / 17) = `156`,

factorial 12 = `479001600`,
choose 2.34 7 = `0.002256232142735999`,
choose (21 / 19) 3 = `-119 / 6859`,
choose 43 30 = `36576848168`,

